ircd/s_newconf.c

```diff
@@ -685,23 +685,39 @@ time_t
 valid_temp_time(const char *p)
 {
 	time_t result = 0;
-
-	while(*p)
-	{
-		if(IsDigit(*p))
-		{
-			result *= 10;
-			result += ((*p) & 0xF);
-			p++;
-		}
-		else
+	int current = 0;
+	char *ch = (char *)p, *och;
```

Comment:
I know it'll make the strtol more cumbersome, but I don't think that's a good enough reason to cast.